### PR TITLE
Happy little entity ID parameterizations

### PR DESCRIPTION
**Pull request by iethree**
### Description
In AuditV2 we're going to mess up, and continue messing up, all entity ids for tests. this PR:
- Adds support for variable entity IDs for dashcards and users
- Updates test hitting the user, card, and dashcard apis

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: e2e/support/helpers/e2e-qa-databases-helpers.js [hunk 1]
```diff
@@ -103,12 +103,12 @@ function assertOnDatabaseMetadata(engine) {
...
 function recursiveCheck(id, i = 0) {
-  // Let's not wait more than 5s for the sync to finish
-  if (i === 20) {
+  // Let's not wait more than 20s for the sync to finish
+  if (i === 40) {
     throw new Error("The sync is taking too long. Something is wrong.");
   }
 
-  cy.wait(250);
+  cy.wait(500);
```
Comment (npfitz) [inline e2e/support/helpers/e2e-qa-databases-helpers.js:111]: Do you know what the reason was for needing to increase the wait time?

File: e2e/test/scenarios/collections/collection-pinned-overview.cy.spec.js [hunk 2]
```diff
@@ -253,6 +256,24 @@ describe("scenarios > collection pinned items overview", () => {
...
+    cy.request("PUT", `/api/card/${ORDERS_QUESTION_ID}`, {
+      collection_position: 1,
+    });
+
+    cy.visit("/collection/root");
+    cy.wait("@getPinnedItems");
+    cy.wait("@getCardQuery");
+
+    cy.findByLabelText("Next page").click();
+    // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
```
Comment (npfitz) [inline e2e/test/scenarios/collections/collection-pinned-overview.cy.spec.js:270]: Not sure we should be adding es lint disables for new lines of code 😅

Comment (npfitz) [reply e2e/test/scenarios/collections/collection-pinned-overview.cy.spec.js:270]: Ahh, looks like it was grabbed from one of the repro's

**Review by npfitz — APPROVED** [review]
Looks good! Just a question and a comment, but mergable as is 👍

File: e2e/support/helpers/e2e-qa-databases-helpers.js (hunk 1 above)
Comment (iethree) [reply e2e/support/helpers/e2e-qa-databases-helpers.js:111]: I've been getting occasional failures, even locally, not just in this branch, sometimes I have to try 2-3 times to get it to sync fast enough. 🤷

Comment (npfitz) [reply e2e/support/helpers/e2e-qa-databases-helpers.js:111]: Could just be that there is more stuff in the snapshot now? Would that make sense?

**Comment by iethree** [comment]
related to https://github.com/metabase/metabase/issues/33571

**Pull request merged**